src/client/cached.go

```diff
@@ -72,7 +74,11 @@ func (d *DiscoveryCacher) discoverAndCache(timeout time.Duration) (HTTPClient, e
 	if err != nil {
 		d.Logger.WithError(err).Warnf("while storing %q in the cache", d.StorageKey)
 	}
-	return client, nil
+	return d.wrap(client, timeout), nil
+}
+
+func expired(creationTimestamp int64, ttl time.Duration) bool {
+	return time.Since(time.Unix(creationTimestamp, 0)) > ttl
 }
```

Comment:
Given that this is now a oneliner, is it worth to have a function wrapper for it?

Reply:
I think it still provides some value with code dedup + naming it. I'd leave it.